app/src/app-logic.js

```diff
@@ -28,34 +30,47 @@ export default function useAppLogic() {
       convictionStakes.filter(({ entity }) => entity === connectedAccount)) ||
     []
 
-  const myCurrentStakes = useMemo(() => {
-    return myStakesHistory.reduce(
-      (stakes, { proposal: currProposalId, tokensStaked }) => {
-        if (tokensStaked === 0) stakes.delete(currProposalId)
-        else {
-          const proposal = proposals.find(({ id }) => id === currProposalId)
-          if (proposal && !proposal.executed) {
-            stakes.set(
-              currProposalId,
-              formatTokenAmount(
-                parseInt(tokensStaked),
-                parseInt(stakeToken.tokenDecimals)
-              )
-            )
-          }
-        }
-        return stakes
-      },
-      new Map()
-    )
-  }, [myStakesHistory.length])
+  const myActiveStakes = useMemo(() => {
+    if (!connectedAccount || !stakeToken.tokenDecimals || !proposals) {
+      return
+    }
+    return proposals.reduce((myStakes, proposal) => {
+      if (proposal.executed || !proposal.stakes) {
+        return myStakes
+      }
+
+      const totalActive = proposal.stakes.reduce((accumulator, stake) => {
+        return accumulator.add(stake.amount)
+      }, new BN('0'))
+
+      setTotalActiveTokens(prevValue => prevValue.add(totalActive))
+
+      const myStake = proposal.stakes.find(
+        stake => stake.entity === connectedAccount
+      )
+      if (!myStake) {
+        return myStakes
+      }
+
+      myStakes.push({
+        proposal: proposal.id,
+        proposalName: proposal.name,
+        stakedAmount: myStake.amount,
+      })
+      setMyActiveTokens(prevValue => prevValue.add(myStake.amount))
+      return myStakes
+    }, [])
+  }, [proposals.length, connectedAccount, stakeToken.tokenDecimals])
+
   const myLastStake = [...myStakesHistory].pop() || []
 
   return {
     onProposalSubmit,
     proposalPanel,
     setProposalPanel,
     myLastStake,
```

Comment:
Looks like this one is not being used anymore? Should we remove it?

Reply:
ah yes!